junit-platform-commons/src/main/java/org/junit/platform/commons/util/CollectionUtils.java

```diff
@@ -139,7 +143,11 @@ public static boolean isConvertibleToStream(Class<?> type) {
 				|| Iterable.class.isAssignableFrom(type)//
 				|| Iterator.class.isAssignableFrom(type)//
 				|| Object[].class.isAssignableFrom(type)//
-				|| (type.isArray() && type.getComponentType().isPrimitive()));
+				|| (type.isArray() && type.getComponentType().isPrimitive())//
+				|| Arrays.stream(type.getMethods())//
+						.filter(m -> m.getName().equals("iterator"))//
```

Comment:
:question: If possible, I think we should check whether `type` implements `kotlin.sequences.Sequence` here because it's safer. We will need to defensively load the Kotlin type via reflection.

Reply:
I followed the same logic, the Kotlin language is following for for-loops, namely that anything that provides an iterator, can be iterated over using a for-loop.  See also https://kotlinlang.org/docs/control-flow.html#for-loops.

I applied this logic to the conversion from anything that provides an iterator to convert to a stream. This also alows a kotlin sequence to be converted to a stream. Don't you think it is safe enough to check wheather the object to convert to a stream contains an iterator providing method with the name 'iterator' and actually provides a java.util.Iterator?

I'm not sure yet how to check via reflection if we are dealing with a kotlin Sequence without having a dependency to the kotlin standard library in the junit-platform-commons module.